authentication/src/main/java/io/camunda/authentication/handler/CustomMethodSecurityExpressionRoot.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.handler;
+
+import io.camunda.authentication.entity.CamundaUser;
+import io.camunda.service.AuthorizationServices;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.security.access.expression.SecurityExpressionRoot;
+import org.springframework.security.access.expression.method.MethodSecurityExpressionOperations;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+public class CustomMethodSecurityExpressionRoot extends SecurityExpressionRoot
+    implements MethodSecurityExpressionOperations {
+
+  private static final Set<String> READ_ACCESS_AUTHORITIES = Set.of(PermissionType.READ.name());
+  private static final Set<String> WRITE_ACCESS_AUTHORITIES =
+      Set.of(
+          PermissionType.UPDATE.name(), PermissionType.CREATE.name(), PermissionType.DELETE.name());
+
+  private final AuthorizationServices<AuthorizationRecord> authorizationServices;
+  private Object filterObject;
+  private Object returnObject;
+
+  public CustomMethodSecurityExpressionRoot(
+      final Authentication authentication,
+      final AuthorizationServices<AuthorizationRecord> authorizationServices) {
+    super(authentication);
+    this.authorizationServices = authorizationServices;
+  }
+
+  public boolean hasReadAccess(final String resourceType) {
+    return hasReadAccess(resourceType, "*");
+  }
+
+  public boolean hasReadAccess(final String resourceType, final String entity) {
+    return hasPermissions(resourceType, entity, READ_ACCESS_AUTHORITIES);
+  }
+
+  public boolean hasWriteAccess(final String resourceType) {
+    return hasWriteAccess(resourceType, "*");
+  }
+
+  public boolean hasWriteAccess(final String resourceType, final String entity) {
+    return hasPermissions(resourceType, entity, WRITE_ACCESS_AUTHORITIES);
```

Comment:
❓ with this approach I would need all permissions that are considered write (CREATE, UPDATE, DELETE) to have access (based on the `.containsAll` check [here](https://github.com/camunda/camunda/pull/23080/files#diff-0880de59b39250a94703e490fd9e900610f62bf91daca18442e1351b90c73093R69)) is that correct or would it be _any_ of those permissions and I have access? In an ideal world its on an endpoint basis so I'm targetting a specific permission but wanted to check

Reply:
It's also one of my confusion point, to be strict write means all this permission because we are not sure if it's going to be a delete or create. I think in reality we will use hasPermissions directly and pass the required permissions to the method.

Reply:
I agree, based on a kick off I attended yesterday we will be more specific with the permission, for now I think I would remove these combined methods and just provide the `hasPermission` method. The others can always be added later on if we need them